tests/conftest.py

```diff
@@ -19,13 +19,27 @@ def table():
             TableName="imap-data-table",
             KeySchema=[
                 # Partition key
-                {"AttributeName": "met", "KeyType": "HASH"},
+                {"AttributeName": "ingest_year", "KeyType": "HASH"},
```

Comment:
This isn't a good partition key, since people will almost always be using the current year and therefore there will be a lot of load on one partition and none on the rest. 

This may or may not even really matter based on your access patterns and data size, just wanted to call it out. 
https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/bp-partition-key-uniform-load.html

Maybe using a hash on the time instead would make sense?

Reply:
Ideally a single partition (like apid) would have been better. Greg and I talked about this. A single DynamoDB partition can store 10 GB of data. We will generate ~1.25 GB/yr. So we would run out of space in 8 years. So I had to break it down. Year seemed to be the least cumbersome way to do that. I could do it by day, but that would require more queries. Time would result in the same problem that I had earlier which is you can't query a range which is likely what the instrument teams will want. I'm open to other ideas though.

Reply:
FYI, I read about this and a single partition does have a 10GB limit internal to AWS, but if you run over that, AWS will start combining the partition key and sort key into the hash and create a **new** partition for you. So it isn't a hard-stop on the 10GB limit.

My suggestion would be to have `apid` be the Partition Key, and `met` be the Sort Key like you initially proposed. Time series queries are much better done along the sort key, and I think it is totally fine to have the table have a single partition key.

Reply:
Oh interesting. Let me read the link you sent.

Reply:
Ok. Updated. Thanks for that info Greg.

Reply:
That sounds good to me, thank you both for your consideration!